helix-term/src/ui/editor.rs

```diff
@@ -771,15 +771,55 @@ impl EditorView {
         cxt: &mut commands::Context,
         event: KeyEvent,
     ) -> Option<KeymapResult> {
+        let mut last_mode = mode;
         let key_result = self.keymaps.get(mode, event);
         cxt.editor.autoinfo = self.keymaps.sticky().map(|node| node.infobox());
 
+        // Track the currently open doc
+        let view = view!(cxt.editor);
+        let doc_id = view.doc;
+
+        let mut execute_command = |command: &commands::MappableCommand| {
+            command.execute(cxt);
+            let doc = cxt.editor.documents.get_mut(&doc_id).unwrap();
```

Comment:
I set remaping for close buffer, It back error.

```
[keys.normal.space]
x = [":buffer-close"]
```

thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', helix-term/src/ui/editor.rs:783:61

Reply:
I added a PR to fix this case: https://github.com/helix-editor/helix/pull/3613

Reply:
when close the last buffer , It will take error:


```diff
thread 'main' panicked at 'invalid HopSlotMap key used', helix-view/src/tree.rs:288:20
stack backtrace:
   0: std::panicking::begin_panic
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/std/src/panicking.rs:616:12
   1: <slotmap::hop::HopSlotMap<K,V> as core::ops::index::IndexMut<K>>::index_mut
             at /Users/erasin/.cargo/registry/src/github.com-1ecc6299db9ec823/slotmap-1.0.6/src/hop.rs:948:21
   2: helix_view::tree::Tree::get_mut
-             at ./helix-view/src/tree.rs:288:20
   3: <helix_term::ui::editor::EditorView as helix_term::compositor::Component>::handle_event
-             at ./helix-term/src/ui/editor.rs:1268:28
   4: helix_term::compositor::Compositor::handle_event
             at ./helix-term/src/compositor.rs:172:19
   5: helix_term::application::Application::handle_terminal_events
             at ./helix-term/src/application.rs:453:22
   6: helix_term::application::Application::event_loop_until_idle::{{closure}}
             at ./helix-term/src/application.rs:292:21
   7: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/core/src/future/mod.rs:91:19
   8: helix_term::application::Application::event_loop::{{closure}}
             at ./helix-term/src/application.rs:268:57
   9: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/core/src/future/mod.rs:91:19
  10: helix_term::application::Application::run::{{closure}}
             at ./helix-term/src/application.rs:838:38
  11: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/4b91a6ea7258a947e59c6522cd5898e7c0a6a88f/library/core/src/future/mod.rs:91:19
  12: hx::main_impl::{{closure}}
             at ./helix-term/src/main.rs:143:53
```


https://github.com/helix-editor/helix/blob/1cbf5525540531f26cd6bd765de93b20d96078f5/helix-term/src/ui/editor.rs#L1254

https://github.com/helix-editor/helix/blob/1cbf5525540531f26cd6bd765de93b20d96078f5/helix-term/src/ui/editor.rs#L1268

I don't know how it works, but it's probably caused by this.

Reply:
There's an issue for this now: https://github.com/helix-editor/helix/issues/3626